generate/lib/write_ts_declarations.js

```diff
@@ -0,0 +1,332 @@
+/**
+ * Consumes the API in JSON format, and produces a TypeScript declaration file.
+ * @author John Vilk <jvilk@cs.umass.edu>
+ */
+var fs = require('fs-extra');
+var Path = require('path');
+
+var writeTsDecls = function(apiData, path) {
+  var path = path || '';
+  path = ("/" + path + "/").replace(/\/+/g, '/');
+
+  /**
+   * Given a type from the API docs, produce a TypeScript type.
+   */
+  function getType(type) {
+    // Remove spaces.
+    type = type.replace(/\s/g, '');
+    // Convert [] to Array<> for generics.
+    var sqBracketIndex = type.indexOf('[');
+    if (sqBracketIndex !== -1) {
+      return "Array<" + getType(type.slice(0, sqBracketIndex)) + ">";
+    }
+
+    // If type has < in it already, translate generic argument.
+    var angleIndex = type.indexOf('<');
+    if (angleIndex !== -1) {
+      type = type.slice(0, angleIndex) + '<' + getType(type.slice(angleIndex + 1, type.indexOf('>'))) + ">";
+    }
+
+    switch (type) {
```

Comment:
Ideally we should eventually move this to a dedicate JSON file like the `descriptor.json`.